### PR TITLE
Use lodash yo

### PR DESCRIPTION
**Pull request by saponifi3d**
Use _.result which will return the value of the function or the value of the object instead of having to check for a function and optionally invoking it.

### CODE REVIEW

**Comment by kylev** [comment]
LGTM

**Comment by jmerrifield** [comment]
Do we need to bump the lodash dependency in package.json to make sure `_.result` is available?

**Comment by saponifi3d** [comment]
@jmerrifield nope. https://github.com/lodash/lodash/blob/2.4.1/doc/README.md#_resultobject-key

**Comment by jmerrifield** [comment]
I think the invocation of `_.result` needs some adjustment

```js
n_ > var o = {val: 123, fn: function() { return 456 }}
undefined
n_ > _.result(o.val)
undefined
n_ > _.result(o.fn)
undefined
n_ > _.result(o, 'val')
123
n_ > _.result(o, 'fn')
456
```

File: index.js [hunk 1]
```diff
@@ -7,10 +7,7 @@ module.exports = function (options) {
     , workers = []
     , restarting = false
 
-  var numWorkers = _.isFunction(options.numWorkers)
-    ? options.numWorkers()
-    : options.numWorkers
-    || 2
+  var numWorkers = _.result(options, numWorkers, 2);
```
Comment (jmerrifield) [inline index.js]: lodash 2.4.1 doesn't have the defaultValue arg, you'd need to update package.json to use this.

Also, quotes around `numWorkers` - needs to be a string

Comment (saponifi3d) [inline index.js]: Ugh I'm done with this. I was trying to be nice... :p

**Comment by jmerrifield** [comment]
Works now that the lodash version has been bumped :)

**Pull request merged**